code/modules/mob/abstract/new_player/new_player.dm

```diff
@@ -323,21 +327,37 @@ INITIALIZE_IMMEDIATE(/mob/abstract/new_player)
 
 	if(unique_role_available)
 		dat += "<font color='[COLOR_BRIGHT_GREEN]'><b>A unique ghost role is available:</b></font><br>"
-	dat += "<a href='byond://?src=\ref[src];ghostspawner=1'>Ghost Spawner Menu</A><br>"
+	dat += "<a href='byond://?src=\ref[src];ghostspawner=1'>Ghost Spawner Menu</A>"
 
 	dat += "Choose from the following open/valid positions:<br>"
+	var/list/list/datum/job/jobs_by_department = list()
+	for(var/department in positions_by_department) // prepare empty list
+		jobs_by_department[department] = list()
 	for(var/datum/job/job in SSjobs.occupations)
 		if(job && IsJobAvailable(job.title))
+			var/department = job.department
+			if(!department || !(department in jobs_by_department)) // no department set or it's something weird
+				department = DEPARTMENT_MISCELLANEOUS
+			if(job.head_position) // make sure heads are first
```

Comment:
See above about department flags. Wouldn't it be nicer to have

```suggestion
#define JOB_IS_HEAD (job.department_flag & HEAD)
...
			if(JOB_IS_HEAD) // make sure heads are first
```

Or something?

Reply:
I'm not sure if this is the best approach - mixing different types of values (jobs and "is head") together. Like, if we wanted to assign Heads "properly" to both their department _and_ command this would cause issues because they are a head in their department but not in Command.

But we definitely need to have just one place to check for these (as well as for departments) otherwise there will be some inconsistency somewhere. I just don't think I should fix that in this PR, it's already ginormous.